components/resources/library/src/commonMain/kotlin/org/jetbrains/compose/resources/ResourceEnvironment.kt

```diff
@@ -0,0 +1,65 @@
+package org.jetbrains.compose.resources
+
+import androidx.compose.foundation.isSystemInDarkTheme
+import androidx.compose.runtime.*
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.text.intl.Locale
+
+internal data class ResourceEnvironment(
+    val language: LanguageQualifier,
+    val region: RegionQualifier,
+    val theme: ThemeQualifier,
+    val density: DensityQualifier
+)
+
+@Composable
+internal fun rememberEnvironment(): ResourceEnvironment {
+    val composeLocale = Locale.current
+    val composeTheme = isSystemInDarkTheme()
+    val composeDensity = LocalDensity.current
+
+    //cache ResourceEnvironment unless compose environment is changed
+    return remember(composeLocale, composeTheme, composeDensity) {
+        ResourceEnvironment(
+            LanguageQualifier(composeLocale.language),
+            RegionQualifier(composeLocale.region),
+            ThemeQualifier.selectByValue(composeTheme),
+            DensityQualifier.selectByDensity(composeDensity.density)
+        )
+    }
+}
+
+/**
+ * Provides the resource environment for non-composable access to string resources.
+ * It is an expensive operation! Don't use it in composable functions with no cache!
+ */
+internal expect fun getResourceEnvironment(): ResourceEnvironment
+
+internal fun Resource.getPathByEnvironment(environment: ResourceEnvironment): String {
```

Comment:
would describe resolution scheme in details and check if it matches Anroid ones (https://developer.android.com/guide/topics/resources/providing-resources#BestMatch). It will help later for documentation

Reply:
My algorithm does the same but a slightly different way. I will add comments